src/solver.rs

```diff
@@ -81,3 +168,111 @@ impl FromStr for SMTSolver {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn parse_counterexample() {
+        for solver in [SolverConfig::new("cvc4"), SolverConfig::new("z3")] {
+            let (sat, eval) = query_smt_with_solver_and_eval(
+                &"(declare-fun x () (_ BitVec 16))(assert (= x #x0100))".to_string(),
+                &vec!["x".to_string()],
+                solver,
+            );
+            assert_eq!(sat, true);
+            assert_eq!(eval, vec![ModelValue::Number(BigUint::from(0x100u32))]);
+        }
+    }
+
+    #[test]
+    fn parse_counterexample_empty_eval() {
+        for solver in [SolverConfig::new("cvc4"), SolverConfig::new("z3")] {
+            let (sat, eval) = query_smt_with_solver_and_eval(
+                &"(declare-fun x () (_ BitVec 16))(assert (= x #x0100))".to_string(),
+                &vec![],
+                solver,
+            );
+            assert_eq!(sat, true);
+            assert_eq!(eval, vec![]);
+        }
+    }
+
+    #[test]
+    fn query_counterexample_unsat() {
+        for solver in [SolverConfig::new("cvc4"), SolverConfig::new("z3")] {
+            let (sat, eval) = query_smt_with_solver_and_eval(
+                &"(declare-fun x () (_ BitVec 16))(assert (and (= x #x0100) (= x #x0000)))"
+                    .to_string(),
+                &vec!["x".to_string()],
+                solver,
+            );
+            assert_eq!(sat, false);
+            assert_eq!(eval, vec![]);
+        }
+    }
+
+    #[test]
+    fn parse_counterexample_multi() {
+        for solver in [SolverConfig::new("cvc4"), SolverConfig::new("z3")] {
```

Comment:
Can also extract the for loop into a function to make sure all tests are using both solvers